lib/elixir/lib/module.ex

```diff
@@ -1864,8 +1900,15 @@ defmodule Module do
         :ets.insert(set, {key, value, line})
         :ets.insert(bag, {:attributes, key})
     else
-      :accumulate -> :ets.insert(bag, {{:accumulate, key}, value})
-      _ -> :ets.insert(set, {key, value, line})
+      :accumulate ->
+        :ets.insert(bag, {{:accumulate, key}, value})
+
+      :unset ->
+        :ets.insert(set, {key, value, line})
+        :ets.insert(bag, {:attributes, key})
```

Comment:
This was what was causing the warning not to happen. It was only inserting the value into the set and not the bag.

Reply:
Good catch. I think an alternative solution would be to write to`{:attributes, key}` table on `register_attribute`, can you please check? I think I prefer to move it there because the best would be to handle `:undef` in as few places as possible.

Reply:
That appears to work. One thing I don't understand is why attributes are tracked under the attributes key in the bag, when you have the attributes in the set?

Reply:
The `attributes` key would be better called `warn_attributes`, because the only purpose is to annotate them for checking for warnings later. A later PR would be appreciated to rename it if you are interested.

The reason we use two of them is because it is the most efficient way to implement this. Traversing set tables in ETS is expensive, so the key-values are stored in the set and in the bag we keep all entries in case we need to traverse.